### PR TITLE
[TK-09573] fix agent_info && move to kitsune_p2p_types

### PR DESCRIPTION
**Pull request by neonphog**
AgentInfoSigned->AgentInfo->AgentMetaInfo has been super un-ergonomic to work with... and requires re-parsing the AgentInfo over and over inefficiently in the code...

This fixes all that, and moves it into the `kitsune_p2p_types` crate in preparation for the rust bootstrap server & binary.

### CODE REVIEW

File: crates/kitsune_p2p/types/src/agent_info.rs [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+    /// Note--the encoded bootstrap version of this struct uses a weird
+    /// offset from the signed time... but this value here is the more
+    /// intuitive absolute value.
+    pub expires_at_ms: u64,
+
+    /// Raw bytes of agent info signature as kitsune signature.
+    pub signature: Arc<KitsuneSignature>,
+
+    /// the raw encoded bytes sent to bootstrap server or use for sig verify.
+    pub encoded_bytes: Box<[u8]>,
```
Comment (freesig) [inline crates/kitsune_p2p/types/src/agent_info.rs:87]: Just curious why a `Box<[u8]>` and not a `Vec<u8>`?

**Review by freesig — APPROVED** [review]
Looks good to me

File: crates/kitsune_p2p/types/src/agent_info.rs (hunk 1 above)
Comment (neonphog) [reply crates/kitsune_p2p/types/src/agent_info.rs:87]: Mostly as a hint that this is static, shouldn't ever grow / shrink.

**Review by maackle — COMMENTED** [review]
Oops, I had a pending review, and never finished/submitted it. I had this one comment which I'll just leave here since it's already written.

**Pull request merged**